Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -529,7 +529,20 @@ void vtkMRMLSliceLogic::ProcessMRMLLogicsEvents()
       }
     else
       {
-      textureToRAS = this->SliceNode->GetXYToRAS();
+      // We use the texture computed by the slice view.
+      // OpenGL texture pixel origin is in the pixel corner, while
+      // in VTK pixel origin is in the pixel center. Therefore, we need to shift
+      // the coordinate by half voxel.
+      // The following code sets textureToRAS = SliceNode->GetXYToRAS() * Translate(-0.5, -0.5, 0)
+      // The translation matrix is almost an identity matrix therefore we the computation can be
+      // easily and more efficiently performed by elementary operations on the matrix elements.
```

Comment:
```suggestion
      // Create a new textureToRAS matrix with translation to correct texture pixel origin
      //
      // Since the OpenGL texture pixel origin is in the pixel corner and the
      // VTK pixel origin is in the pixel center, we need to shift the coordinate
      // by half voxel.
      //
      // Considering that the translation matrix is almost an identity matrix, the 
      // computation easily and efficiently performed by elementary operations on
      // the matrix elements.
```